indexing-service/src/main/java/org/apache/druid/indexing/common/task/BatchAppenderators.java

```diff
@@ -59,13 +63,21 @@ public static Appenderator newAppenderator(
       TaskToolbox toolbox,
       DataSchema dataSchema,
       AppenderatorConfig appenderatorConfig,
+      FirehoseFactory firehoseFactory,
       DataSegmentPusher segmentPusher
   )
   {
+    final boolean isReingest;
+    if (firehoseFactory instanceof IngestSegmentFirehoseFactory) {
+      isReingest = dataSchema.getDataSource().equals(((IngestSegmentFirehoseFactory) firehoseFactory).getDataSource());
+    } else {
+      isReingest = false;
+    }
```

Comment:
is it possible to drive this from auto compaction code in druid coordinator instead as `IngestSegmentFirehoseFactory` could be used outside of auto compaction as well. For example, as a user , knowing my data flow, I can setup re-index task to run every day for previous day's data .. sort of a manual compaction. But in that case, the `CompactionState` doesn't need to be preserved.

Reply:
Hmm, I thought `lastCompactionState` could be useful even for manual compaction as well. How about adding a parameter to taskContext to store `lastCompactionState`, so that other users also can use it if they want?

Reply:
yeah , that would work and let user (in this case auto compaction code) explicitly say whether CompactionState should be saved or not.

Reply:
Thanks, added a new task context configuration. I'm still not sure whether this should be documented though.

Reply:
For now, I think this config is best left undocumented and for auto compaction internal usage only.